### PR TITLE
More universal startup script.

### PR DESCRIPTION
**Pull request by paul-1**
Add auto detection of touch drivers.
Add 90 and 270 rotations for the portrait mode piDisplay 2.
Detect if vc4 or firmware drivers are being used.
Move all environment variables to this script.

### CODE REVIEW

**Comment by paul-1** [comment]
@blaisedias  could you give this script a try on your systems.

File: jivelite-sp [hunk 1]
```diff
@@ -4,22 +4,57 @@ export LOG=/var/log/pcp_jivelite.log
...
-        fi
-    fi	
 fi
 
+# Autodetect Touchscreen, and then mouse if no touch is found.
+TMPFILE=$(mktemp)
+udevadm info --export-db | grep "P: " | grep "event" | sed 's/P: //' > $TMPFILE
+for LINE in $(cat $TMPFILE); do
+   udevadm info --query=property --path=$LINE | grep -q TOUCH
+   [ $? -eq 0 ] && eventno=$(basename $LINE);TOUCH=1;MOUSE=0
```
Comment (blaisedias) [inline jivelite-sp]: Could we do something like this?
``` 
  [ "$eventno" != "" ] &&  echo "Found Touch screen: $eventno" >> $LOG && break
```

File: jivelite-sp [hunk 2]
```diff
@@ -4,22 +4,57 @@ export LOG=/var/log/pcp_jivelite.log
...
+   [ "$eventno" != "" ] && break
+   udevadm info --query=property --path=$LINE | grep -q MOUSE
+   [ $? -eq 0 ] && eventno=$(basename $LINE);TOUCH=0;MOUSE=1
```
Comment (blaisedias) [inline jivelite-sp]: similarly
```
   [ "$eventno" != "" ] && echo "Found mouse: $eventno" >> $LOG && break
```

File: jivelite-sp [hunk 3]
```diff
@@ -4,22 +4,57 @@ export LOG=/var/log/pcp_jivelite.log
...
     export SDL_FBDEV=$JL_FRAME_BUFFER
     echo "Using $SDL_FBDEV as frame buffer device." >> $LOG
+
+    case $(cat /proc/fb | grep -E ^${JL_FRAME_BUFFER: -1} | cut -d' ' -f2) in
+        vc4drmfb|drm-rp1-dsidrmf) DRIVER="VC4";;
+        BCM2708) DRIVER="BCM2708";;
+        *) DRIVER="OTHER"
+    esac
+fi
+
```
Comment (blaisedias) [inline jivelite-sp:68]: and also
```
echo "driver = $DRIVER" >> $LOG
```

**Review by blaisedias — COMMENTED** [review]
@paul-1 works on my PI 0.
I made some minor suggestions so that the log files would contain more information

File: jivelite-sp (hunk 1 above)
Comment (paul-1) [reply jivelite-sp]: I added a bit more information rather than just the even number

**Review by blaisedias — APPROVED** [review]
LGTM

**Pull request merged**